packages/SwingSet/docs/dynamic-vats.md

```diff
@@ -0,0 +1,99 @@
+# Dynamic Vats
+
+A SwingSet machine has two sets of vats. The first are the *Static Vats*, which are defined in the configuration object, and created at startup time. The root objects of these vats are made available to the `bootstrap()` method, which can wire them together as it sees fit.
+
+The second set are the *Dynamic Vats*. These are created after startup, with source code bundles that were now known at boot time. Typically these bundles arrive over the network from external providers. In the Agoric system, contracts are installed as dynamic vats (each spawned instance goes into a separate vat).
+
+Dynamic vats are metered: each message delivered gets a limited amount of resources (CPU cycles, memory, stack frames). If it exceeds this budget, the vat is terminated. All outstanding messages will be rejected, any future messages will be rejected, and it will never get CPU time again.
+
+## Creating a Dynamic Vat
+
+Vats are created by sending a `createVat()` message to the *Vat Admin Service* object, containing the source bundle which defines your new vat. Dynamic vats can only be created by other vats on the same SwingSet, so ultimately one of the static vats must cooperate.
+
+The ability to create new vats is not ambient: it requires access to the Vat Admin Service object, which is initially only made available to the bootstrap call. The bootstrap call might choose to share it with other objects, or not.
+
+### Making the Source Bundle
+
+The first step is to create a source bundle. To do this, you'll want to point the `bundleSource` function at a local source file. This file should have a function named `buildRootObject` as it's default export. Suppose your vat code is stored in `vat-counter.js`:
```

Comment:
nit: s/it's/its/

Reply:
Also, it's not "as its default export", it should be "as a named export."

Reply:
Good point, I assumed we had to stick with the default export for now to be compatible with existing dynamic vat code, but.. there aren't any yet, only tests. So now is the perfect time to change. I'll switch the docs and the implementation to look for an export named `buildRootObject`.

(contracts continue to use a named export of `makeContract`.. when zoe starts putting contract instances into their own vats, it won't be the contract that initially populates the vat, it'll be the ZCF. So the ZCF code will export `buildRootObject` because that's what vats do, and the ZCF code will do `ns = await importBundle(contract); contract = ns.makeContract()`)

Reply:
Code updated in https://github.com/Agoric/agoric-sdk/pull/1232 , which should feed into this branch.